Geometry/fieldFactory_func.f90

```diff
@@ -0,0 +1,80 @@
+!!
+!! Module to build new fields. Called by geometryReg_mod
+!!
+module fieldFactory_func
+
+  use numPrecision
+  use genericProcedures, only : fatalError
+  use dictionary_class,  only : dictionary
+
+  ! Fields
+  use field_inter,              only : field
+  use uniformScalarField_class, only : uniformScalarField
+  use uniformVectorField_class, only : uniformVectorField
+  use uniFissSitesField_class,  only : uniFissSitesField
+
+  ! Geometry
+  use geometryReg_mod,          only : gr_addField => addField
+
+  implicit none
+  private
+
+
+  !! Parameters
+  character(nameLen), dimension(*), parameter :: AVAILABLE_FIELDS = ['uniformScalarField',&
+                                                                     'uniformVectorField',&
+                                                                     'uniFissSitesField ']
+
+   ! Public interface
+   public :: new_field
+
+contains
+
+  !!
+  !! Allocates and initialises field from dictionary
+  !!
+  !! Args:
+  !!   dict [in]   -> Dictionary with definition
+  !!   name [in]   -> Name of the field for the geometry registry
+  !!
+  !! Errors:
+  !!   fatalError is type of field is unknown
+  !!
+  subroutine new_field(dict, name)
```

Comment:
Maybe it is worth to explicitly mention that the field gets allocated inside the `geometryReg`?